src/wordlift/mappings/class-mappings-dbo.php

```diff
@@ -241,20 +251,34 @@ public function get_rule_group_list( $mapping_id ) {
 	/**
 	 * Gets a list of rule group items.
 	 *
-	 * @param int $mapping_id Primary key for mapping table.
+	 * @param string $mapping_id Primary key for mapping table.
 	 *
 	 * @return array Get list of rule group items.
 	 */
 	public function get_rule_groups_by_mapping( $mapping_id ) {
-
+		$mapping_id            = (string) $mapping_id;
 		$rule_group_table_name = $this->wpdb->prefix . WL_RULE_GROUP_TABLE_NAME;
-		$rule_group_rows       = $this->wpdb->get_results(
-			$this->wpdb->prepare(
-				"SELECT rule_group_id FROM $rule_group_table_name WHERE mapping_id=%d",
-				$mapping_id
-			),
-			ARRAY_A
-		);
+		$rule_group_rows       = array();
+		// Do a integer check, because the mapping_id might be string value too
+		// If it is a string then we can directly proceed to filter mappings.
+		if ( is_numeric( $mapping_id ) ) {
+			$rule_group_rows = $this->wpdb->get_results(
+				$this->wpdb->prepare(
+					"SELECT rule_group_id FROM $rule_group_table_name WHERE mapping_id=%d",
+					(int) $mapping_id
+				),
+				ARRAY_A
+			);
+		}
+
+		$manual_rule_groups = apply_filters( 'wl_mappings_register_rule_groups', array() );
```

Comment:
@naveen17797 pls

- [ ] add PhpDoc for the filter

- [ ] rename the filter to `wl_mappings_rule_groups`

- [ ] preset the `array()` to `$rule_group_rows` so that hooks can also modify existing values, and you can remove the `array_merge` at ln 280.

- [ ] add the  `$mapping_id` parameter in the filter so that hooks can provide rule groups for the specific mapping id and you can remove the `array_filter` at ln 276.